lib/IRGen/IRGenSIL.cpp

```diff
@@ -3567,8 +3568,11 @@ void IRGenSILFunction::visitRefTailAddrInst(RefTailAddrInst *i) {
 }
 
 static bool isInvariantAddress(SILValue v) {
-  auto root = getUnderlyingAddressRoot(v);
-  if (auto ptrRoot = dyn_cast<PointerToAddressInst>(root)) {
+  SILValue accessedAddress = getAccessedAddress(v);
+  if (accessedAddress->getType().isAddress() && isLetAddress(accessedAddress)) {
+    return true;
+  }
+  if (auto ptrRoot = dyn_cast<PointerToAddressInst>(accessedAddress)) {
```

Comment:
`auto *`

Reply:
Not part of this commit, but sure I'll fix it.